src/components/emailSelector/Label.tsx

```diff
@@ -147,7 +149,7 @@ const Label = ({
           break;
 
         case BACKSPACE:
-          if (!event.repeat && content === '' && onRemove) {
+          if (!event.repeat && content === '' && onRemove && !((index === 0 || index === 1) && disableRemovalOfFirst)) {
             onRemove(index);
```

Comment:
Why not make that check part of the onRemove function?

Reply:
Because the check is different for BACKSPACE & normal remove. The backpace checks also index 1 and the normal doesnt.